### PR TITLE
Add documentation for adding sidecar injector webhook annotations in 1.4

### PR DESCRIPTION
**Pull request by jacob-delgado**
Add documentation for installing istio 1.4.8 with istioctl. Due to https://github.com/istio/istio/pull/20949 not being backported to istio 1.4, but changes to support OpenShift 4.3.5 (https://github.com/istio/istio/issues/22449) documentation is required for installing.

We cannot update the archive for istio 1.4.

We can just update 1.5 or merge into master and cherry-pick into 1.5.

### CODE REVIEW

File: content/en/docs/setup/additional-setup/cni/index.md [hunk 1]
```diff
@@ -136,6 +136,24 @@ The following table shows the required settings for many common Kubernetes envir
...
+
+Istioctl in 1.4.8+ does not support [escaping strings](https://github.com/istio/istio/issues/19196). To install on
+OpenShift 4.2+ you will need to create a file and pass that into istioctl:
+
+cni-annotations.yaml
+```yaml
+global:
+  sidecarInjectorWebhook:
+    injectedAnnotations:
+      "k8s.v1.cni.cncf.io/networks": istio-cni
```
Comment (jacob-delgado) [inline content/en/docs/setup/additional-setup/cni/index.md:149]: I wasn't able to get this to be applied. Tried various other settings like

```yaml
sidecarInjectorWebhook:
  injectedAnnotations:
    "k8s.v1.cni.cncf.io/networks": istio-cni
```

File: content/en/docs/setup/additional-setup/cni/index.md [hunk 2]
```diff
@@ -136,6 +136,33 @@ The following table shows the required settings for many common Kubernetes envir
...
+Istioctl in 1.4.8+ does not support [escaping strings](https://github.com/istio/istio/issues/19196). To install on
+OpenShift 4.2+ you will need to create a file and pass that into istioctl:
+
+cni-annotations.yaml
+```yaml
+apiVersion: install.istio.io/v1alpha1
+kind: IstioOperator
+spec:
+  components:
+    cni:
```
Comment (ericvn) [inline content/en/docs/setup/additional-setup/cni/index.md:150]: {{< text yaml >}}
cat <<EOF> cni-annotations.yaml
apiVersion: install.istio.io/v1alpha1
kind: IstioOperator
spec:
  components:
    cni:
      enabled: true
      namespace: kube-system
    values:
      cni:
        chained: false
        cniBinDir: /var/lib/cni/bin
        cniConfDir: /etc/cni/multus/net.d
        cniConfFileName: istio-cni.conf
      sidecarInjectorWebhook:
        injectedAnnotations:
          "k8s.v1.cni.cncf.io/networks": istio-cni
EOF
{{< /text >}}

Comment (ericvn) [reply content/en/docs/setup/additional-setup/cni/index.md:150]: Well, I though I submitted a change suggestion. Trying again. Maybe it doesn't work in this case.

File: content/en/docs/setup/additional-setup/cni/index.md [hunk 3]
```diff
@@ -136,6 +136,33 @@ The following table shows the required settings for many common Kubernetes envir
 | Red Hat OpenShift 3.10+ | _(none)_ | _(none)_ |
 | Red Hat OpenShift 4.2+ | `--set components.cni.namespace=kube-system --set values.cni.cniBinDir=/var/lib/cni/bin --set values.cni.cniConfDir=/etc/cni/multus/net.d --set values.cni.chained=false --set values.cni.cniConfFileName="istio-cni.conf" --set values.sidecarInjectorWebhook.injectedAnnotations."k8s\.v1\.cni\.cncf\.io/networks"=istio-cni` | _(none)_ |
 
+#### Istioctl support in 1.4.8+ for OpenShift 4.2+
```
Comment (jwendell) [inline content/en/docs/setup/additional-setup/cni/index.md:139]: I'm not good with English, but I would say something like `Instructions for Istio 1.4 and OpenShift`.

Then I'd add a note stating that Docs for 1.4 are outdated.

File: content/en/docs/setup/additional-setup/cni/index.md [hunk 4]
```diff
@@ -136,6 +136,33 @@ The following table shows the required settings for many common Kubernetes envir
...
+  values:
+    cni:
+      chained: false
+      cniBinDir: /var/lib/cni/bin
+      cniConfDir: /etc/cni/multus/net.d
+      cniConfFileName: istio-cni.conf
+    sidecarInjectorWebhook:
+      injectedAnnotations:
+        "k8s.v1.cni.cncf.io/networks": istio-cni
+```
```
Comment (ericvn) [inline content/en/docs/setup/additional-setup/cni/index.md:162]: {{< text yaml >}}
cat < cni-annotations.yaml
apiVersion: install.istio.io/v1alpha1
kind: IstioOperator
spec:
components:
cni:
enabled: true
namespace: kube-system
values:
cni:
chained: false
cniBinDir: /var/lib/cni/bin
cniConfDir: /etc/cni/multus/net.d
cniConfFileName: istio-cni.conf
sidecarInjectorWebhook:
injectedAnnotations:
"k8s.v1.cni.cncf.io/networks": istio-cni
EOF
{{< /text >}}

File: content/en/docs/setup/additional-setup/cni/index.md [hunk 5]
```diff
@@ -136,6 +136,33 @@ The following table shows the required settings for many common Kubernetes envir
 | Red Hat OpenShift 3.10+ | _(none)_ | _(none)_ |
 | Red Hat OpenShift 4.2+ | `--set components.cni.namespace=kube-system --set values.cni.cniBinDir=/var/lib/cni/bin --set values.cni.cniConfDir=/etc/cni/multus/net.d --set values.cni.chained=false --set values.cni.cniConfFileName="istio-cni.conf" --set values.sidecarInjectorWebhook.injectedAnnotations."k8s\.v1\.cni\.cncf\.io/networks"=istio-cni` | _(none)_ |
 
+#### Istioctl support in 1.4.8+ for OpenShift 4.2+
+
+Istioctl in 1.4.8+ does not support [escaping strings](https://github.com/istio/istio/issues/19196). To install on
```
Comment (jwendell) [inline content/en/docs/setup/additional-setup/cni/index.md:141]: I'd rephrase that to something like `Due to a limitation in istioctl 1.4, you need to create a file and pass that file to istioctl`.

File: content/en/docs/setup/additional-setup/cni/index.md (hunk 2 above)
Comment (ericvn) [reply content/en/docs/setup/additional-setup/cni/index.md:150]: Anyway, the thought was to replace that snippet with a text yaml (and get the single line requirement correct) so one can copy and paste to get the file created. This also removed the file name spelling issue.

Then the file is used on the next line.

File: content/en/docs/setup/additional-setup/cni/index.md [hunk 6]
```diff
@@ -136,6 +136,33 @@ The following table shows the required settings for many common Kubernetes envir
...
+      chained: false
+      cniBinDir: /var/lib/cni/bin
+      cniConfDir: /etc/cni/multus/net.d
+      cniConfFileName: istio-cni.conf
+    sidecarInjectorWebhook:
+      injectedAnnotations:
+        "k8s.v1.cni.cncf.io/networks": istio-cni
+```
+
+Then passing in the file in as an argument `-f cni-annotations.yaml`
```
Comment (ericvn) [inline content/en/docs/setup/additional-setup/cni/index.md:164]: I'm making an assumption that people want to add more to their command lines than just this flag, else you could use a text bash to make something like `istioctl manifest generate -f cni-annotations.yaml` or whatever form the command is to simplify cut/paste.

File: content/en/docs/setup/additional-setup/cni/index.md (hunk 5 above)
Comment (jacob-delgado) [reply content/en/docs/setup/additional-setup/cni/index.md]: I like that a lot better.

File: content/en/docs/setup/additional-setup/cni/index.md (hunk 3 above)
Comment (jacob-delgado) [reply content/en/docs/setup/additional-setup/cni/index.md]: Done.

File: content/en/docs/setup/additional-setup/cni/index.md (hunk 6 above)
Comment (jacob-delgado) [reply content/en/docs/setup/additional-setup/cni/index.md]: Hopefully I made it more clear that other variables can be set or changed when doing an install. Need further review.

File: content/en/docs/setup/additional-setup/cni/index.md [hunk 7]
```diff
@@ -136,6 +136,35 @@ The following table shows the required settings for many common Kubernetes envir
 | Red Hat OpenShift 3.10+ | _(none)_ | _(none)_ |
 | Red Hat OpenShift 4.2+ | `--set components.cni.namespace=kube-system --set values.cni.cniBinDir=/var/lib/cni/bin --set values.cni.cniConfDir=/etc/cni/multus/net.d --set values.cni.chained=false --set values.cni.cniConfFileName="istio-cni.conf" --set values.sidecarInjectorWebhook.injectedAnnotations."k8s\.v1\.cni\.cncf\.io/networks"=istio-cni` | _(none)_ |
 
+#### Instructions for Istio 1.4.x and OpenShift
+
+Due to a [limitation](https://github.com/istio/istio/issues/19196) in `istioctl` 1.4.x, you will need to create a file and
+pass that file to `istioctl`. The documentation for Istio 1.4.x is out of date for OpenShift installation.
+
+{{< text yaml >}}
+cat <<'EOF' > cni-annotations.yaml
```
Comment (jwendell) [inline content/en/docs/setup/additional-setup/cni/index.md:145]: Did you actually try these instructions? I get this error: `could not unmarshal the overlay file: unknown field "components" in v1alpha2.IstioControlPlaneSpec`

This file works with `istioctl manifest`:
```yaml
apiVersion: install.istio.io/v1alpha1
kind: IstioOperator
spec:
  cni:
    enabled: true
    components:
      cni:
        namespace: kube-system
  values:
    cni:
      chained: false
      cniBinDir: /var/lib/cni/bin
      cniConfDir: /etc/cni/multus/net.d
      cniConfFileName: istio-cni.conf
    sidecarInjectorWebhook:
      injectedAnnotations:
        "k8s.v1.cni.cncf.io/networks": istio-cni
```

File: content/en/docs/setup/additional-setup/cni/index.md [hunk 8]
```diff
@@ -136,6 +136,35 @@ The following table shows the required settings for many common Kubernetes envir
...
+      cniBinDir: /var/lib/cni/bin
+      cniConfDir: /etc/cni/multus/net.d
+      cniConfFileName: istio-cni.conf
+    sidecarInjectorWebhook:
+      injectedAnnotations:
+        "k8s.v1.cni.cncf.io/networks": istio-cni
+EOF
+{{< /text >}}
+
+Then passing in the file in as an argument to istioctl: `-f cni-annotations.yaml`. The use of `--set` with custom
```
Comment (jwendell) [inline content/en/docs/setup/additional-setup/cni/index.md:165]: What about this:

Then pass in this file as an argument to `istioctl`, for example: `istioctl manifest apply -f cni-annotations.yaml`. You can still pass other command line arguments with `--set` if you need to.

I'd also put a warning like this:
```
    {{< warning >}}
    In order to deploy Istio 1.4 on OpenShift with CNI you need to use at least Istio 1.4.8.
    {{< /warning >}}
```

**Comment by jwendell** [comment]
@jacob-delgado You also need to backport https://github.com/istio/istio/pull/21421

File: content/en/docs/setup/additional-setup/cni/index.md (hunk 7 above)
Comment (jacob-delgado) [reply content/en/docs/setup/additional-setup/cni/index.md:145]: My mistake. Turns out I had istioctl 1.5 in my path. Updated. I'll verify again before it goes live.

File: content/en/docs/setup/additional-setup/cni/index.md [hunk 9]
```diff
@@ -136,6 +136,48 @@ The following table shows the required settings for many common Kubernetes envir
...
+variables can be appended to your use of istioctl.
+
+Then pass in this file as an argument to istioctl, for example:
```
Comment (jwendell) [inline content/en/docs/setup/additional-setup/cni/index.md:169]: You forgot to delete the section above, thus this is duplicate. Also, guard `istioctl` with `. You can see the preview at https://deploy-preview-7101--preliminary-istio.netlify.app/

File: content/en/docs/setup/additional-setup/cni/index.md [hunk 10]
```diff
@@ -136,6 +136,45 @@ The following table shows the required settings for many common Kubernetes envir
...
+EOF
+{{< /text >}}
+
+Then pass in this file as an argument to `istioctl`, for example:
+
+{{< text bash >}}
+$ istioctl manifest apply -f cni-annotations.yaml
+{{< /text >}}
+
+You can still pass other command line arguments with `--set` if you need to.
```
Comment (frankbu) [inline content/en/docs/setup/additional-setup/cni/index.md:172]:
```suggestion
You can pass other command line arguments with `--set` if necessary.
```

**Comment by jacob-delgado** [comment]
I consent

File: content/en/docs/setup/additional-setup/cni/index.md [hunk 11]
```diff
@@ -136,6 +136,45 @@ The following table shows the required settings for many common Kubernetes envir
 | Red Hat OpenShift 3.10+ | _(none)_ | _(none)_ |
 | Red Hat OpenShift 4.2+ | `--set components.cni.namespace=kube-system --set values.cni.cniBinDir=/var/lib/cni/bin --set values.cni.cniConfDir=/etc/cni/multus/net.d --set values.cni.chained=false --set values.cni.cniConfFileName="istio-cni.conf" --set values.sidecarInjectorWebhook.injectedAnnotations."k8s\.v1\.cni\.cncf\.io/networks"=istio-cni` | _(none)_ |
 
+#### Instructions for Istio 1.4.x and OpenShift
+
+Due to a [limitation](https://github.com/istio/istio/issues/19196) in `istioctl` 1.4.x, you will need to create a file and
+pass that file to `istioctl`. The documentation for Istio 1.4.x is out of date for OpenShift installation.
```
Comment (frankbu) [inline content/en/docs/setup/additional-setup/cni/index.md:142]:
```suggestion
Due to a [limitation](https://github.com/istio/istio/issues/19196) in `istioctl` 1.4.x, CNI will not work properly with the
default installation of Istio on OpenShift. To work around this problem, create the following YAML file:
```

File: content/en/docs/setup/additional-setup/cni/index.md [hunk 12]
```diff
@@ -136,6 +136,45 @@ The following table shows the required settings for many common Kubernetes envir
...
+      cniBinDir: /var/lib/cni/bin
+      cniConfDir: /etc/cni/multus/net.d
+      cniConfFileName: istio-cni.conf
+    sidecarInjectorWebhook:
+      injectedAnnotations:
+        "k8s.v1.cni.cncf.io/networks": istio-cni
+EOF
+{{< /text >}}
+
+Then pass in this file as an argument to `istioctl`, for example:
```
Comment (frankbu) [inline content/en/docs/setup/additional-setup/cni/index.md:166]:
```suggestion
Then pass this file as an argument to `istioctl`, for example:
```

**Comment by jacob-delgado** [comment]
I consent
